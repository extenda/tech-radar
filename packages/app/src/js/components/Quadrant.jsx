import PropTypes from 'prop-types';
import React from 'react';

import radarService from '../modules/radarService';
import Navigation from './Navigation';
import NotFound from './NotFound';
import QuadrantList from './QuadrantList';

const Quadrant = ({ match }) => {
  const quadrant = radarService.getQuadrant(match.params.quadrant);

  if (!quadrant) {
    return <NotFound />;
  }

  return (
    <>
      <Navigation quadrant={{ name: quadrant.name, dirname: quadrant.dirname }} />
      <div className="container">
        <h1>{quadrant.name}</h1>
        <QuadrantList quadrant={quadrant} />
      </div>
    </>
  );
};

Quadrant.propTypes = {
  match: PropTypes.shape({
    params: PropTypes.shape({
      quadrant: PropTypes.string.isRequired,
    }).isRequired,
  }).isRequired,
};

export default Quadrant;
