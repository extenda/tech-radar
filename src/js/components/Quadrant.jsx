import React, { Component } from 'react';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';
import radarService from '../modules/radarService';

export default class Quadrant extends Component {
  static propTypes = {
    match: PropTypes.shape({
      params: PropTypes.shape({
        quadrant: PropTypes.string.isRequired,
      }).isRequired,
    }).isRequired,
  };

  renderEntriesList = (quadrant, ring) => (
    <ul className={ring}>
      { quadrant[ring].map(entry => (
        <li key={entry.filename}>
          <Link to={`/entries/${entry.filename}`}>{entry.name}</Link>
        </li>
      ))}
    </ul>
  );

  render = () => {
    const { match } = this.props;
    const quadrant = radarService.getQuadrant(match.params.quadrant);
    return (
      <div className="container">
        <h1>{quadrant.name}</h1>
        <div className="row">
          <div className="one-half column">
            <h2>Adopt</h2>
            {this.renderEntriesList(quadrant, 'adopt')}

            <h2>Trial</h2>
            {this.renderEntriesList(quadrant, 'trial')}
          </div>
          <div className="one-half column">
            <h2>Assess</h2>
            {this.renderEntriesList(quadrant, 'assess')}

            <h2>Hold</h2>
            {this.renderEntriesList(quadrant, 'hold')}
          </div>
        </div>
      </div>
    );
  };
}
