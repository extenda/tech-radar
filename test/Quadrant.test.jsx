import React from 'react';
import { shallow } from 'enzyme';
import Quadrant from '../src/js/components/Quadrant';

jest.mock('../src/js/modules/radarService');

describe('<Quadrant />', () => {
  test('It groups entries by ring', () => {
    const match = { params: { quadrant: 'test' } };
    const component = shallow(<Quadrant match={match} />);

    expect(component.find('ul.adopt > li')).toHaveLength(2);
    expect(component.find('ul.trial > li')).toHaveLength(0);
    expect(component.find('ul.assess > li')).toHaveLength(1);
    expect(component.find('ul.hold > li')).toHaveLength(3);
  });

  test('It renders as expected', () => {
    const match = { params: { quadrant: 'test' } };
    const component = shallow(<Quadrant match={match} />);
    expect(component).toMatchSnapshot();
  });
});
