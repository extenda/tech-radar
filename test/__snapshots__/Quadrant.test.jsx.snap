// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Quadrant /> It renders as expected 1`] = `
<Fragment>
  <Navigation
    entry={null}
    home={null}
    quadrant={
      {
        "dirname": "test1",
        "name": "Test",
      }
    }
    tag={null}
  />
  <div
    className="container"
  >
    <h1>
      Test
    </h1>
    <QuadrantList
      blips={
        [
          {
            "id": -1,
          },
        ]
      }
      headerLevel={2}
      quadrant={
        {
          "adopt": [
            {
              "active": true,
              "filename": "1.html",
              "id": 1,
              "name": "A - 1",
            },
            {
              "active": true,
              "filename": "2.html",
              "id": 2,
              "name": "B - 2",
            },
          ],
          "assess": [
            {
              "active": true,
              "filename": "3.html",
              "id": 3,
              "name": "C - 3",
            },
          ],
          "dirname": "test1",
          "hold": [
            {
              "active": true,
              "filename": "4.html",
              "id": 4,
              "name": "D - 4",
            },
            {
              "active": true,
              "filename": "5.html",
              "id": 5,
              "name": "E - 5",
            },
            {
              "active": false,
              "filename": "6.html",
              "id": 6,
              "name": "F - 6",
            },
          ],
          "name": "Test",
          "trial": [],
        }
      }
      useShortname={false}
    />
  </div>
</Fragment>
`;

exports[`<Quadrant /> It renders not found for invalid quadrant 1`] = `<NotFound />`;
